README.md

```diff
@@ -1,63 +1,109 @@
-  [Gherkin]: https://github.com/cucumber/cucumber/wiki/Gherkin
-  [Dredd]: https://github.com/apiaryio/dredd
-  [hooks]: https://dredd.readthedocs.io/en/latest/hooks/
+[Dredd](https://dredd.org) is a tool for testing web APIs. It supports [hooks](http://dredd.org/en/latest/hooks/index.html) written in [many languages](http://dredd.org/en/latest/hooks/index.html#supported-languages). To support a particular language, it needs an adapter, so-called hooks handler. This [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development) test suite ensures that the public interface of each hooks handler works as Dredd expects. The test suite is written in [Gherkin](https://github.com/cucumber/cucumber/wiki/Gherkin) and uses [Cucumber](https://github.com/cucumber/cucumber-js) as a test runner.
 
 ## Usage
 
-### Install the features dependencies
+First read the docs about [how to create a new hooks handler](http://dredd.org/en/latest/hooks/new-language.html) for Dredd. Implement your hooks handler and then make sure it passes this test suite.
 
-```bash
-npm install -g dredd
-git clone https://github.com/apiaryio/dredd-hooks-template.git
-cd dredd-hooks-template
-bundle install
-```
+### Installation
+
+1.  Make sure you have [Node.js](https://nodejs.org/) and [npm](https://www.npmjs.com/package/npm) available.
+1.  Create a `package.json` file in the root of your project. This is where your JavaScript dependencies are going to be specified:
+
+    ```json
+    {
+      "scripts": { "test": "dredd-hooks-template test" },
+      "private": true
+    }
+    ```
+
+1.  Run `npm install dredd-hooks-template --save-dev` to install and declare this test suite as your development dependency.
+1.  Run `npx dredd-hooks-template init` to get a copy of the test suite in the `./features` directory.
+1.  Open the feature files in `./features/*.feature` and in all of them
+
+    -   replace `dredd-hooks-{{mylanguage}}` with a path to your hooks handler executable which you want to get tested (e.g. `./bin/dredd_hooks`)
+    -   replace `{{myextension}}` by the extension of the hooks handler language (e.g. `.py`),
+    -   uncomment the code blocks and rewrite them to the hooks handler language.
+
+Now you have the test suite ready.
+
+### Usage
 
-### Enable the features for your language
+Every time you run `npx dredd-hooks-template test` (or `npm test`), you should see the test suite running. The end goal is that all the tests pass:
 
-1. Open the feature files in `./features/*.feature`
-1. In all of them, replace:
-  - `{{mylanguage}}` by the hooks handler command for you language
-  - `{{myextension}}` by the extension for your language
-1. Implement the code examples in your language
-1. Run the test suite: `bundle exec cucumber`
+![test suite passing](passing.png)
+
+You should add the `package.json` file to Git. When starting from scratch, you can run `npm install` to install the JavaScript dependencies.
+
+<a name="upgrading"></a>
```

Comment:
Does this anchor do something, or is it a leftover?

Reply:
It's a trick to ensure the following (at the bottom of the README file) works regardless the thing which renders the markdown:

```markdown
1.  Go to Python hooks repository and [upgrade](#upgrading) the test suite package there. Remove any local `steps.js` as the necessary steps should already be implemented in the new version of the test suite package.
1.  Go to Ruby hooks repository and [upgrade](#upgrading) the test suite package there.
```